mirage/ramp_generator/obs_generator.py

```diff
@@ -197,10 +197,10 @@ def add_crs_and_noise(self, seed, num_integrations=None):
                 ngroups = self.params['Readout']['ngroup']
             # TSO and moving targets
             elif seeddim == 4:
-                if num_integrations != seed.shape[0]:
-                    raise ValueError(('The number of integrations reported for the dark ({}) does not match that '
-                                      'implied by the size of the 1st dimension of the seed image ({}). Not sure '
-                                      'which to use.'.format(num_integrations, seed.shape[0])))
+                # if num_integrations != seed.shape[0]:
```

Comment:
Did you mean to leave this commented out?

Reply:
This `ValueError` gets thrown for anything over 213 integrations if I leave it in but the mismatch seems to be corrected somewhere along the line if I comment it out, i.e. any number of integrations work if this `raise` statement is not here! I have NO idea why.

Reply:
Very strange. I've run NIRCam TSO exposures with more than 213 integrations and have never run across this. Is this a case where the seed image is being split into multiple files? Maybe that's why the number of integrations pulled from the dark file header isn't matching the size of the seed image? If file splitting is happening, then Mirage works with one exposure segment at a time, so NINT in the dark will be less than the total number of integrations in the exposure. Do you know if SOSS observations will undergo file splitting? Mirage is set up to split observations with > 160 full frame images (i.e. groups x ints). But I recently learned that DMS is only applying this rule to some observing modes. So Mirage is different than DMS since it will split e.g. NIRCam imaging data as well, but with a limit of 160 frames, which is hard to reach in imaging mode, fixing this has been low on the priority list.

Reply:
Yes, that's what happening. I was running a 420 integration simulation and it split it into 213 and 207 int chunks but then breaks when 213 (the first chunk) doesn't match 420. There will be some pretty long SOSS simulations for sure. Should I leave this commented out or does it break something else?

Reply:
Yes it does!